frappe/utils/jinja.py

```diff
@@ -77,15 +77,40 @@ def render_template(template, context, is_path=None, safe_render=True):
 		return ""
 
 	if (is_path or guess_is_path(template)):
-		return get_jenv().get_template(template).render(context)
+		rendered = get_jenv().get_template(template).render(context)
+		add_preload_hints(rendered)
+		return rendered
 	else:
 		if safe_render and ".__" in template:
 			throw("Illegal template")
 		try:
-			return get_jenv().from_string(template).render(context)
+			rendered = get_jenv().from_string(template).render(context)
+			add_preload_hints(rendered)
+			return rendered
 		except TemplateError:
 			throw(title="Jinja Template Error", msg="<pre>{template}</pre><pre>{tb}</pre>".format(template=template, tb=get_traceback()))
 
+def add_preload_hints(content):
```

Comment:
You can do this in `build_page` after all of the HTML has been rendered:

https://github.com/frappe/frappe/blob/9c7f01b59d69799eaae5354f49c44d67c87f6091/frappe/website/render.py#L184